src/Controller/BlogPostController.php

```diff
@@ -0,0 +1,207 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\DTO\BlogPostEnquiry;
+use App\DTO\BlogPostListEnquiry;
+use App\Entity\BlogPost;
+use App\Entity\BlogUser;
+use App\Repository\BlogUserRepository;
+use App\Service\BlogPostService;
+use App\Service\Serializer\DTOSerializer;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Exception\NotSupported;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\ORM\ORMException;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\BlogPostRepository;
+
+class BlogPostController extends AbstractController
+{
+
+    /**
+     * @Route("/blogposts/{userID?}", name="blog-posts", methods={"GET"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     * returns blogposts by userID or If userID null, all blogposts
+     */
+    public function getBlogPostList(
+        Request $request,
+        EntityManagerInterface $entityManager,
+        DTOSerializer $serializer,
```

Comment:
[Jeff]
- `serializer` looks better to declare in constructor
- `userID` usually stored in a session, don't pass params